### ISSUE TITLE
Public STUN servers

### ISSUE THREAD

**Issue by heiher** [heiher/natmap#18]
## TCP/UDP

* stun.stunprotocol.org (Not available now)
* stun.voip.blackberry.com
* stun.nextcloud.com

## UDP Only

* stun.qq.com

**Comment by MikeWang000000** [comment]
stun.stunprotocol.org -> stunserver.stunprotocol.org

ref: https://stunprotocol.org

**Comment by xqzr** [comment]
```log
stun.byterace.shop: Non-existent domain
```

**Comment by MikeWang000000** [comment]
以下均收集自各 WebRTC 服务商提供的 TURN 服务器，可当作 TCP STUN 服务器使用：

```
w1.xirsys.com
u1.xirsys.com
relay1.expressturn.com
relay2.expressturn.com
relay3.expressturn.com
relay4.expressturn.com
relay5.expressturn.com
relay6.expressturn.com
relay7.expressturn.com
relay8.expressturn.com
global.turn.twilio.com
stun.relay.metered.ca:80
stun.kinesisvideo.us-east-2.amazonaws.com:443
stun.kinesisvideo.us-east-1.amazonaws.com:443
stun.kinesisvideo.us-west-2.amazonaws.com:443
stun.kinesisvideo.af-south-1.amazonaws.com:443
stun.kinesisvideo.ap-east-1.amazonaws.com:443
stun.kinesisvideo.ap-south-1.amazonaws.com:443
stun.kinesisvideo.ap-northeast-2.amazonaws.com:443
stun.kinesisvideo.ap-southeast-1.amazonaws.com:443
stun.kinesisvideo.ap-southeast-2.amazonaws.com:443
stun.kinesisvideo.ap-northeast-1.amazonaws.com:443
stun.kinesisvideo.ca-central-1.amazonaws.com:443
stun.kinesisvideo.eu-central-1.amazonaws.com:443
stun.kinesisvideo.eu-west-1.amazonaws.com:443
stun.kinesisvideo.eu-west-2.amazonaws.com:443
stun.kinesisvideo.eu-west-3.amazonaws.com:443
stun.kinesisvideo.sa-east-1.amazonaws.com:443
```

**Comment by IcedMango** [comment]
`stunserver.stunprotocol.org` seems to be not working.

[Global DNS lookup](https://www.whatsmydns.net/#A/stunserver.stunprotocol.org) shows it points to 127.0.0.1.

<img width="628" alt="image" src="https://github.com/heiher/natmap/assets/22059332/85041baa-4888-47d3-82b1-b4305d629ee7">

**Comment by napbonking** [comment]
stunserver2024.stunprotocol.org

**Comment by wray-lee** [comment]
stun.hot-chilli.net
stun.yy.com

**Comment by github-xihe** [comment]
stunserver2025.stunprotocol.org

**Comment by WofWca** [comment]
I believe the maintainer of stunprotocol.org doesn't really want it to be used by big production projects:
https://github.com/jselbie/stunserver/issues/50#issuecomment-1428416301

**Comment by heiher** [comment]
> I believe the maintainer of stunprotocol.org doesn't really want it to be used by big production projects: [jselbie/stunserver#50 (comment)](https://github.com/jselbie/stunserver/issues/50#issuecomment-1428416301)

Yeah. Earlier, @MikeWang000000 had already pointed this out (https://github.com/heiher/natmap/issues/18#issuecomment-1445352795), and soon after we lowered the priority of `stunprotocol.org` in the table before eventually removing it altogether.

**Comment by xxxxxliil** [comment]
某位在 https://www.v2ex.com/t/1037362 的五楼贴了一个 gits：https://gist.github.com/mondain/b0ec1cf5f60ae726202e
在 go-nats 的 readme.md 搜一个 stun 服务器域名发现的两个 gits：
https://gist.github.com/sagivo/3a4b2f2c7ac6e1b5267c2f1f59ac6c6b
https://gist.github.com/SujithManjavana/44603c80335deabf22a755a41a7da8fc
在 go-stun 的 issue 里发现的另外一个 gits：https://gist.github.com/zziuni/3741933

还能在这些 gits 看到，而且也是最应该在这里提到的一个项目：https://github.com/pradt2/always-online-stun
……感觉像是在 kde 的一大堆软件里找自己想用的一样，那里面真有坏的

**Comment by q741451** [comment]
国内访问总会有很多奇怪的问题，要么不支持udp，要么经常连不上。如果有公网条件，可以快速搭建一个：https://github.com/q741451/stunsrv
